### PR TITLE
I made a task for the Java classes section and added a condition before the entry point

### PR DESCRIPTION
**Pull request by AleksandrVoyteshenko**
…re the entry point

### CODE REVIEW

File: src/javaclasses/TheDatabaseOfTheInstitution.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                new Student("6", "Шаповалов", "Павел", "Денисович", "05.01.1996", "Могилев", "375446666666", "Энергетический", "2", "2-19"),
+                new Student("7", "Белун", "Евгений", "Леонидович", "02.01.1996", "Гродно", "375447777777", "Энергетический", "2", "2-19"),
+                new Student("8", "Амонов", "Роман", "Брониславович", "18.04.1994", "Витебск", "375448888888", "Правоведение", "2", "2-19"),
+                new Student("9", "Кривошеев", "Руслан", "Сергеевич", "15.01.2000", "Лида", "375449999999", "Юридический", "3", "3-18"),
+                new Student("10", "Рыгало", "Николай", "Степанович", "21.07.1999", "Мозырь", "375440000000", "Юридический", "3", "3-18")
+        };
+
+        Student[] studentsList = studentsArray;
+
+        Scanner g = new Scanner(System.in);
```
Comment (nitrogen16) [inline src/javaclasses/TheDatabaseOfTheInstitution.java:38]: разнести логические блоки по отдельным методам

File: src/javaclasses/TheDatabaseOfTheInstitution.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            for (Student st : studentsList) {
+                if (st.getFaculty().equals(faculty)) {
+                    System.out.println(st.toString());
+                } else {
+                    ;
+                }
+            }
+        }
+
+        Scanner y = new Scanner(System.in);
```
Comment (nitrogen16) [inline src/javaclasses/TheDatabaseOfTheInstitution.java:66]: каждый раз содавать объект Scanner? а зачем, если можно использовать один

**Review by nitrogen16 — COMMENTED** [review]
исправить комментарии

File: src/java_classes/TheDatabaseOfTheInstitution.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        System.out.println("Список студентов родившихся после " + year + "года");
+        for (Student st : studentsArray) {
+            if (parseYear(st.getDateOfBirth().substring(6)) > parseYear(year)) {
+                System.out.println(st);
+            }
+        }
+    }
+
+    static void listOfStudentByGourse(Student[] studentsArray) {
+
```
Comment (nitrogen16) [inline src/java_classes/TheDatabaseOfTheInstitution.java:82]: extra space

File: src/java_classes/Student.java [hunk 4]
```diff
@@ -0,0 +1,123 @@
+package java_classes;
+
+public class Student {
+
+    private String id;
```
Comment (nitrogen16) [inline src/java_classes/Student.java:5]: id should be int

File: src/java_classes/TheDatabaseOfTheInstitution.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        System.out.println("Список студентов факультета " + faculty);
+        for (Student st : studentsArray) {
+            if (st.getFaculty().equals(faculty)) {
+                System.out.println(st);
+            }
+        }
+    }
+
+    static void listOfStudentByYear(Student[] studentsArray) {
+
```
Comment (nitrogen16) [inline src/java_classes/TheDatabaseOfTheInstitution.java:70]: extra space

File: src/java_classes/TheDatabaseOfTheInstitution.java [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+        System.out.println("\nВведите год рождения: ");
+        String year = scanner().nextLine();
+        System.out.println("Список студентов родившихся после " + year + "года");
+        for (Student st : studentsArray) {
+            if (parseYear(st.getDateOfBirth().substring(6)) > parseYear(year)) {
```
Comment (nitrogen16) [inline src/java_classes/TheDatabaseOfTheInstitution.java:75]: substring is not a good approach
it works with 20-05-2020
but if we have 1-1-2020?
use DateFormatter
The idea is to work with Date, not with a String. Therefore, parse String to Date, then get a year from this Date

**Review by nitrogen16 — CHANGES_REQUESTED** [review]
Fix the comments.

File: src/java_classes/TheDatabaseOfTheInstitution.java [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+                new Student(10, "Рыгало", "Николай", "Степанович", "21.07.1999", "Мозырь", "375440000000", "Юридический", "3", "3-18")
+        };
+        listOfStudentByFaculty(studentsArray);
+        listOfStudentByGourse(studentsArray);
+        listOfStudentByYear(studentsArray);
+        listOfStudentsByGroupNumber(studentsArray);
+    }
+
+    static Scanner scanner() {
+        Scanner input = new Scanner(System.in);
```
Comment (nitrogen16) [inline src/java_classes/TheDatabaseOfTheInstitution.java:45]: Each time you call method a new instance of Scanner is created.
Why? Make ONE instance of Scanner for all method calls.

**Review by nitrogen16 — CHANGES_REQUESTED** [review]
Fix the comment. Then I'll approve

File: src/java_classes/TheDatabaseOfTheInstitution.java (hunk 7 above)
Comment (AleksandrVoyteshenko) [reply src/java_classes/TheDatabaseOfTheInstitution.java]: fixed the project structure, left one copy of the scanner for all methods

**Review by nitrogen16 — APPROVED** [review]
More or less it's okay

**Pull request closed without merging**